include/sigmatch/sigmatch.hpp

```diff
@@ -950,7 +951,10 @@ namespace impl {
 //
 class msvc_c7599_false_positive_workaround : public std::span<const std::byte>
 {
+public:
     using std::span<const std::byte>::span;
+    
+    msvc_c7599_false_positive_workaround(const std::span<const std::byte>& span) : std::span<const std::byte>(span) {}
```

Comment:
Regarding the second error. Before, if we call the constructors of the base class `std::span` from a derived class, MSVC will always raise a C7599 false positive error. Although my workaround can suppress that false positive error, it is not a legal C++ code (will raise the error you said).

So the constructor you added is correct. But since the whole C7599 false positive bug has been fixed, I think we can now remove the class `impl::msvc_c7599_false_positive_workaround` and your constructor should be added directly to `const_byte_span`, and it would be nice to add specifiers (`constexpr`, `noexcept`) and documentation on it :)

Suggested changes:

```patch
 include/sigmatch/sigmatch.hpp | 41 +++++++++++------------------------
 1 file changed, 13 insertions(+), 28 deletions(-)

diff --git a/include/sigmatch/sigmatch.hpp b/include/sigmatch/sigmatch.hpp
index ab5cd99..aadf339 100644
--- a/include/sigmatch/sigmatch.hpp
+++ b/include/sigmatch/sigmatch.hpp
@@ -938,31 +938,6 @@ class platform_string
     }
 };
 
-///
-/// @cond
-///
-
-namespace impl {
-
-//
-// Inheriting from `std::span` directly will cause MSVC (ver <= v19.29) false positive C7599 error
-//
-// TODO: Remove this class when MSVC releases fixes.
-//
-class msvc_c7599_false_positive_workaround : public std::span<const std::byte>
-{
-public:
-    using std::span<const std::byte>::span;
-    
-    msvc_c7599_false_positive_workaround(const std::span<const std::byte>& span) : std::span<const std::byte>(span) {}
-};
-
-} // namespace impl
-
-///
-/// @endcond
-///
-
 //////////////////////////////////////////////////
 ///
 /// @brief An extended const byte span class
@@ -970,10 +945,20 @@ class msvc_c7599_false_positive_workaround : public std::span<const std::byte>
 /// In contrast to `std::span<const std::byte>`, it can accept pure pointers and `std::span<const
 /// unsigned char>`.
 ///
-class const_byte_span : public impl::msvc_c7599_false_positive_workaround
+class const_byte_span : public std::span<const std::byte>
 {
 public:
-    using impl::msvc_c7599_false_positive_workaround::msvc_c7599_false_positive_workaround;
+    using std::span<const std::byte>::span;
+
+    ///
+    /// @brief Constructor for `std::span<const std::byte>`
+    ///
+    /// @param[in] range A `std::span<const std::byte>` value.
+    ///
+    constexpr const_byte_span(const std::span<const std::byte> &range) noexcept
+        : std::span<const std::byte>{range}
+    {
+    }
 
     ///
     /// @brief Constructor for pure pointer
@@ -982,7 +967,7 @@ class const_byte_span : public impl::msvc_c7599_false_positive_workaround
     /// @param[in] size  The size of this memory range.
     ///
     constexpr const_byte_span(const void *begin, size_t size) noexcept
-        : impl::msvc_c7599_false_positive_workaround{static_cast<const std::byte *>(begin), size}
+        : std::span<const std::byte>{static_cast<const std::byte *>(begin), size}
     {
     }
```